system/HTTP/URI.php

```diff
@@ -1025,11 +1023,7 @@ protected function parseStr(string $query): array
         $return = [];
         $query  = explode('&', $query);
 
-        $params = array_map(static function (string $chunk) {
-            return preg_replace_callback('/^(?<key>[^&=]+?)(?:\[[^&=]*\])?=(?<value>[^&=]+)/', static function (array $match) {
-                return str_replace($match['key'], bin2hex($match['key']), $match[0]);
-            }, urldecode($chunk));
-        }, $query);
+        $params = array_map(static fn (string $chunk) => preg_replace_callback('/^(?<key>[^&=]+?)(?:\[[^&=]*\])?=(?<value>[^&=]+)/', static fn (array $match) => str_replace($match['key'], bin2hex($match['key']), $match[0]), urldecode($chunk)), $query);
```

Comment:
Can you break this into many lines?

Reply:
updated https://github.com/codeigniter4/CodeIgniter4/pull/5462/commits/dd95ae72be1b6f9a7019cd919fd94446d19a0757